lib/view_component/instrumentation.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ViewComponent # :nodoc:
+  module Instrumentation
+    class << self
+      def included(mod)
+        mod.prepend(self)
+      end
+
+      def enabled?
+        Rails.application.config.view_component.instrumentation_enabled
+      end
+    end
+
+    def render_in(view_context, &block)
+      return super(view_context, &block) unless Instrumentation.enabled?
+
+      ActiveSupport::Notifications.instrument("render.view_component", identifier: self.class.short_identifier) do
```

Comment:
What do you think about using the class name here instead of the path?

```suggestion
      ActiveSupport::Notifications.instrument("render.view_component", identifier: self.class.name) do
```

Reply:
Should we follow the Rails pattern here?

https://github.com/rails/rails/blob/722303b4a77307c85b988793a7aa299b1f478aca/actionview/lib/action_view/template.rb#L357-L359

```ruby
{ virtual_path: @virtual_path, identifier: @identifier }
```

Reply:
I decided to add both `name` and full `identifier`